### PR TITLE
Extensions update

### PR DESCRIPTION
**Pull request by VALERA771**
* Added `ItemExtension::GetCategory(ItemType)`
* Removed `RoleExtension::GetTeam(RoleTypeId)` cuz NW has same method

### CODE REVIEW

**Comment by xNexus-ACS** [comment]
breaking change

**Comment by VALERA771** [comment]
Any ways to remove it?

**Comment by louis1706** [comment]
> breaking change

Where ??

**Comment by xNexus-ACS** [comment]
> > breaking change
> 
> 
> 
> Where ??

Bcs of puttin GetTeam() Extension in obsolete

**Comment by VALERA771** [comment]
> > > breaking change
> > 
> > 
> > 
> > Where ??
> 
> Bcs of puttin GetTeam() Extension in obsolete

But why it's breaking? I think if remove it yeah. But not putting it obsolete

**Comment by louis1706** [comment]
> > > breaking change
> > 
> > 
> > Where ??
> 
> Bcs of puttin GetTeam() Extension in obsolete

that not a breaking change for already exiting plugin

File: Exiled.API/Extensions/ItemExtensions.cs [hunk 1]
```diff
@@ -308,5 +308,35 @@ public static uint GetBaseCode(this FirearmType type)
...
+                return ItemCategory.Ammo;
+            if (type.IsKeycard())
+                return ItemCategory.Keycard;
+            if (type.IsArmor())
+                return ItemCategory.Armor;
+            if (type.IsMedical())
+                return ItemCategory.Medical;
+            if (type.IsWeapon())
+                return ItemCategory.Firearm;
+            if (type.IsScp())
```
Comment (louis1706) [inline Exiled.API/Extensions/ItemExtensions.cs:334]: Scp500 is medical but not an in Scp Category

**Review by louis1706 — CHANGES_REQUESTED** [review]

File: Exiled.API/Extensions/ItemExtensions.cs (hunk 1 above)
Comment (VALERA771) [reply Exiled.API/Extensions/ItemExtensions.cs:334]: Medical is above so if Scp500 is medical it'll return medical category

Comment (louis1706) [reply Exiled.API/Extensions/ItemExtensions.cs:334]: yeah that why is not good

Comment (VALERA771) [reply Exiled.API/Extensions/ItemExtensions.cs:334]: I think doing it with comparing ItemType will be not good

**Comment by xNexus-ACS** [comment]
> > > > breaking change
> > > 
> > > 
> > > Where ??
> > 
> > 
> > Bcs of puttin GetTeam() Extension in obsolete
> 
> that not a breaking change for already existing plugin

if a plugin is using that extension yes it is, at least it will not throw errors when using it lmao

**Review by louis1706 — APPROVED** [review]

**Review by Misaka-ZeroTwo — APPROVED** [review]

File: Exiled.API/Extensions/RoleExtensions.cs [hunk 2]
```diff
@@ -60,6 +60,7 @@ public static class RoleExtensions
         /// </summary>
         /// <param name="roleType">The <see cref="RoleTypeId"/>.</param>
         /// <returns><see cref="Team"/>.</returns>
+        [Obsolete("Use PlayerRolesUtils.GetTeam(roleType) in NW-API", false)]
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Extensions/RoleExtensions.cs:63]:
```suggestion
        [Obsolete("Use PlayerRolesUtils.GetTeam(roleType) instead", false)]
```
my bad, it's in the assembly

**Review by Misaka-ZeroTwo — APPROVED** [review]

**Review by louis1706 — APPROVED** [review]

**Review by Misaka-ZeroTwo — COMMENTED** [review]
You can remove GetTeam now.

**Comment by louis1706** [comment]
DoNotMerge will be keep until every break change are removed (or than we make a major update)

**Review by louis1706 — APPROVED** [review]

**Comment by Thundermaker300** [comment]
I don't see the problem with keeping our own version of `GetTeam()`? As long as it uses the base-game method under-the-hood. That's similar to what the majority of our API is anyways lol.

**Comment by Misaka-ZeroTwo** [comment]
fix build errors.

**Review by Thundermaker300 — APPROVED** [review]

**Pull request merged**